modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -178,10 +199,34 @@ private void _validate(ObjectEntry objectEntry) {
 						"r_apiApplicationToAPIEndpoints_c_apiApplicationId"))) {
 
 				throw new ObjectEntryValuesException.InvalidObjectField(
-					"An API endpoint must be related to a an API application",
+					"An API endpoint must be related to an API application",
 					"an-api-endpoint-must-be-related-to-an-api-application",
 					null);
 			}
+
+			if (((long)values.get(
+					"r_requestAPISchemaToAPIEndpoints_c_apiSchemaId") != 0) &&
```

Comment:
Maybe more readable and better to avoid getting twice each id.

```
long requestAPISchemaId = (long)values.get(
	"r_requestAPISchemaToAPIEndpoints_c_apiSchemaId");

if (requestAPISchemaId != 0 && !_isAPISchema(requestAPISchemaId)) {
	throw new ObjectEntryValuesException.InvalidObjectField(
		"An API endpoint must be related to a valid API schema",
		"an-api-endpoint-must-be-related-to-a-valid-api-schema",
		null);
}
```

Reply:
And what whould happen if an user only changes one of the `schemaIds` to an invalid one , I think that we should validate both of them isn't it? @ccorreagg

Reply:
Yes! That is correct. I meant moving the casting into a variable assignment and them, avoid casting twice for each validation